### PR TITLE
build: add formatting check to the build; also hooks for developers who want pre-commit hook for those

### PR DESCRIPTION
**Pull request by mswertz**
Main feature: 
* build will now check format (spotlessCheck and formatCheck/prettier for java and javascript respectively)
* build will break on format error

In addition optionally, two pre-commit hooks that can be installed if you like
* either automatic formatting of java (spotlessApply) and javascript/vue (format)
* or if you prefer to fix yourself only check formatting  (spotlessCheck) and javascript/vue (checkFormat)

To install the hooks use one of the following:

```
./gradlew installPreCommitGitFormatCheckHook
git commit -m "empty commit"
```

and

```
./gradlew installPreCommitGitFormatApplyHook
git commit -m "empty commit"
```

### CODE REVIEW

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/cohorts/[cohort]/index.vue [hunk 1]
```diff
@@ -137,14 +137,16 @@ const variables = { pid: route.params.cohort };
...
-    baseURL: config.public.apiBase,
-    method: "POST",
-    body: { query, variables },
-  }
-);
+const {
+  data: cohortData,
+  pending,
+  error,
+  refresh,
```
Comment (connoratrug) [inline apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/cohorts/[cohort]/index.vue:144]: does not improve readability

File: apps/nuxt3-ssr/utils/fingerprint.js [hunk 2]
```diff
@@ -1 +1 @@
-export const hash = "";
+export const hash = "lfsqqzav";
```
Comment (connoratrug) [inline apps/nuxt3-ssr/utils/fingerprint.js:1]: this should not be committed

File: gradle.properties [hunk 3]
```diff
@@ -1,4 +1,5 @@
 org.gradle.parallel=true
+org.gradle.caching=true
```
Comment (connoratrug) [inline gradle.properties:2]: what does this do ?

Comment (mswertz) [reply gradle.properties:2]: In theory it helps caching also when your gralde daemon is not yet started, however, I must say I don't see much difference. So I will remove for now.

**Review by connoratrug — APPROVED** [review]

**Pull request merged**